code-styling/markdown/README.md

```diff
@@ -0,0 +1,1155 @@
+  
+  Neither is this header
+  ----------------------
+  ```
+
+- Headers MUST NOT be longer then 60 characters. Longer headers MUST be re-evaluated.
+
+- Headers MUST be preceded and followed by a newline. There is on exception,
+  the header at the start of the document MUST NOT be preceded by a newline.
+
+- Headers MUST NOT increase by more than one "level" from one header to the next.
+
+  _Correct example_:
+  
+  ```Markdown
+  # First level header
+  
+  ## Next level header
+  ```
+  
+  _Incorrect example_:
+  
+  ```Markdown
+  # First level header
+  
+  ### Next level header
+  ```
+
+- The first letter of a header MUST be upper case, unless it is a word that always starts with lowercase letters.
+
+- The other letters SHOULD have the same case they would have in the middle of a sentence.
+
+  _Correct examples_:
+  
+  ```Markdown
+  # This is a correct example of a header text
+  ```
+  
+  ```Markdown
+  # int main
+  ```
+  
+  _Incorrect example_:
+  
+  ```Markdown
+  # This is an Incorrect Example
+  ```
+
+- Headers MUST NOT end with a punctuation character (`.`, `,`, `!`, `?`, `:`, `;`)
+
+  _Correct example_:
+  
+  ```Markdown
+  # How to install this software
+  ```
+
+  _Incorrect examples_:
+  
+  ```Markdown
+  # How to install this software:
+  ```
+
+  ```Markdown
+  # Hello world!
+  ```
+
+- Numbered headers MUST be followed by a point (`.`) character after the number.
+
+  _Correct example_:
+  
+  ```Markdown
+  ### 3.2.1. This header is numbered correctly
+  ```
+  
+  _Incorrect example_:
+  
+  ```Markdown
+  ### 3.2.1 This header is missing a point
+  ```
+
+- Headers define sections. A sections MUST NOT be empty.
+
+  _Correct example_:
+  
+  ```Markdown
+  ### 3.2.1. This it the first section
+  
+  Lorem ipsum is great for this purpose!
+  
+  ### 3.2.2. The start of the second section
+  ```
+
+  _Incorrect example_:
+  
+  ```Markdown
+  ### 3.2.1. This it the first section
+
+  ### 3.2.2. The start of the second section
+  ```
+  
+## 3. Emphasis
+
+- Bold text MUST use the double asterisk (`**`) format.
+  
+  _Correct example_:
+
+  ```Markdown
+  **This text is printed in bold!**
+  ```
+  
+  _Incorrect example_:
+  
+  ```Markdown
+  __This text is also printed in bold...__
+  ```
+
+- Italic text MUST use the underscore (`_`) format.
+  
+  _Correct example_:
+
+  ```Markdown
+  _This text is printed in italic!_
+  ```
+  
+  _Incorrect example_:
+  
+  ```Markdown
+  *This text is also printed in italic...*
+  ```
+
+- Striketrough MUST NOT be used.
+
+- Uppercase text for emphasis SHOULD NOT be used, use emphasis constructs like **bold** or _italic_ instead.
+  The following keywords are excepted from this rule (as defined in [RFC 2119][rfc2119]):
+  > "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+  > "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+
+  _Correct example_:
+  
+  ```Markdown
+  *Note*: One SHALL NOT emphasis in uppercase.
```

Comment:
This example is actually incorrect. It should use either a single underscore to indicate italic, or double asterisk to indicate bold.